test/functional/legacy/082_spec.lua

```diff
@@ -0,0 +1,105 @@
+    execute([[    call Ch(a:a, '>?', a:b, 0)]])
+    execute('  endif')
+    execute('endfunction')
+    execute('function! Check(a, b, result)')
+    execute('  call Chk(a:a, a:b, a:result)')
+    execute('  call Chk(a:b, a:a, -a:result)')
+    execute('endfunction')
+    execute('function! LT(a, b)')
+    execute('  call Check(a:a, a:b, -1)')
+    execute('endfunction')
+    execute('function! GT(a, b)')
+    execute('  call Check(a:a, a:b, 1)')
+    execute('endfunction')
+    execute('function! EQ(a, b)')
+    execute('  call Check(a:a, a:b, 0)')
+    execute('endfunction')
+    execute('let b:passed=0')
+    execute([[call EQ('', '')]])
+    execute([[call LT('', 'a')]])
+    execute([[call EQ('abc', 'abc')]])
+    execute([[call EQ('Abc', 'abC')]])
+    execute([[call LT('ab', 'abc')]])
+    execute([[call LT('AB', 'abc')]])
+    execute([[call LT('ab', 'aBc')]])
+    execute([[call EQ('\xd0\xb9\xd1\x86\xd1\x83\xd0\xba\xd0\xb5\xd0\xbd', '\xd0\xb9\xd0\xa6\xd0\xa3\xd0\xba\xd0\x95\xd0\xbd')]])
+    execute([[call LT('\xd0\xb9\xd1\x86\xd1\x83\xd0\xba\xd0\xb5\xd0\xbd', '\xd0\xaf\xd1\x86\xd1\x83\xd0\xba\xd0\xb5\xd0\xbd')]])
+    execute([[call EQ('\xe2\x84\xaa', 'k')]])
+    execute([[call LT('\xe2\x84\xaa', 'kkkkkk')]])
+    execute([[call EQ('\xe2\x84\xaa\xe2\x84\xaa\xe2\x84\xaa', 'kkk')]])
+    execute([[call LT('kk', '\xe2\x84\xaa\xe2\x84\xaa\xe2\x84\xaa')]])
+    execute([[call EQ('\xe2\x84\xaa\xe2\x84\xa6k\xe2\x84\xaak\xcf\x89', 'k\xcf\x89\xe2\x84\xaakk\xe2\x84\xa6')]])
+    execute([[call EQ('Abc\x80', 'AbC\x80')]])
+    execute([[call LT('Abc\x80', 'AbC\x81')]])
+    execute([[call LT('Abc', 'AbC\x80')]])
+    -- Case folding stops at the first bad character.
+    execute([[call LT('abc\x80DEF', 'abc\x80def')]])
+    execute([[call LT('\xc3XYZ', '\xc3xyz')]])
+    -- FF3A (upper), FF5A (lower).
+    execute([[call EQ('\xef\xbc\xba', '\xef\xbd\x9a')]])
+    -- First string is ok and equals \xef\xbd\x9a after folding, second string is illegal and was left unchanged, then the strings were bytewise compared.
+    execute([[call GT('\xef\xbc\xba', '\xef\xbc\xff')]])
+    execute([[call LT('\xc3', '\xc3\x83')]])
+    execute([[call EQ('\xc3\xa3xYz', '\xc3\x83XyZ')]])
+    execute([[for n in range(0x60, 0xFF) | call LT(printf('xYz\x%.2X', n-1), printf('XyZ\x%.2X', n)) | endfor]])
+    execute([[for n in range(0x80, 0xBF) | call EQ(printf('xYz\xc2\x%.2XUvW', n), printf('XyZ\xc2\x%.2XuVw', n)) | endfor]])
+    execute([[for n in range(0xC0, 0xFF) | call LT(printf('xYz\xc2\x%.2XUvW', n), printf('XyZ\xc2\x%.2XuVw', n)) | endfor]])
+    execute([[call append(0, printf('%d checks passed', b:passed))]])
+
+    -- Test that g~ap changes one paragraph only.
+    execute('new')
+    feed('iabcd<cr><cr>')
+    feed('defg<esc>gg0g~ap:let lns = getline(1,3)<cr>')
+    execute('q!')
+    execute([[call append(line('$'), lns)]])
+
+    -- Assert buffer contents.
+    expect([=[
+3732 checks passed
+
+ABCD
+
+defg]=])
```

Comment:
I had trouble with this as the indent must be preset on the empty lines as well but I delete trailing WS with an autocmd similar to this:
`autocmd bufwrite *  %s/\s\+$//e`. Now I have the autocmd disabled manually.